src/main/java/com/github/mtsochacki/adventofcode/Day18.java

```diff
@@ -0,0 +1,183 @@
+        return reduceSlug(resultSlug);
+    }
+
+    private List<RegularNumber> addSnailfish(List<RegularNumber> firstSnailfish, List<RegularNumber> secondSnailfish) {
+        firstSnailfish.addAll(secondSnailfish);
+        for (RegularNumber slug : firstSnailfish) {
+            slug.incrementLevel();
+        }
+        return firstSnailfish;
+    }
+
+    private List<RegularNumber> reduceSlug(List<RegularNumber> snailfishNumber) {
+        List<RegularNumber> reducedSnailfish = copySnailfishNumber(snailfishNumber);
+        while (isInNeedOfExplosion(reducedSnailfish) || isInNeedOfSplit(reducedSnailfish)) {
+            if (isInNeedOfExplosion(reducedSnailfish)) {
+                explode(reducedSnailfish);
+            } else if (isInNeedOfSplit(reducedSnailfish)) {
+                split(reducedSnailfish);
+            }
+        }
+        return reducedSnailfish;
+    }
+
+    private boolean isInNeedOfExplosion(List<RegularNumber> snailfishNumber) {
+        return snailfishNumber.stream()
+                .anyMatch(regularNumber -> regularNumber.getLevel() > 4);
+    }
+
+    private boolean isInNeedOfSplit(List<RegularNumber> snailfishNumber) {
+        return snailfishNumber.stream()
+                .anyMatch(regularNumber -> regularNumber.getValue() > 9);
+    }
+
+    private void explode(List<RegularNumber> snailfishNumber) {
+        for (int i = 0; i < snailfishNumber.size(); i++) {
+            if (snailfishNumber.get(i).getLevel() >= 5) {
+                if (i == 0) {
+                    snailfishNumber.get(i).setValue(0);
+                    snailfishNumber.get(i).decrementLevel();
+                    snailfishNumber.get(i + 2).increaseValue(snailfishNumber.get(i + 1).getValue());
+                    snailfishNumber.remove(i + 1);
+                } else if (i + 2 >= snailfishNumber.size()) {
+                    snailfishNumber.get(i - 1).increaseValue(snailfishNumber.get(i).getValue());
+                    snailfishNumber.get(i).setValue(0);
+                    snailfishNumber.get(i).decrementLevel();
+                    snailfishNumber.remove(i + 1);
+                } else {
+                    snailfishNumber.get(i - 1).increaseValue(snailfishNumber.get(i).getValue());
+                    snailfishNumber.get(i + 2).increaseValue(snailfishNumber.get(i + 1).getValue());
+                    snailfishNumber.get(i + 1).setValue(0);
+                    snailfishNumber.get(i + 1).decrementLevel();
+                    snailfishNumber.remove(i);
+                }
+                break;
+            }
+        }
+    }
+
+    private void split(List<RegularNumber> snailfishNumber) {
+        for (int i = 0; i < snailfishNumber.size(); i++) {
+            if (snailfishNumber.get(i).getValue() > 9) {
+                int value = snailfishNumber.get(i).getValue();
+                int level = snailfishNumber.get(i).getLevel() + 1;
+                snailfishNumber.remove(i);
```

Comment:
Why remove and add instead of overwriting the value?

Reply:
done

Reply:
done